src/renderer/protocol-handler/app-handlers.ts

```diff
@@ -0,0 +1,34 @@
+import { addClusterURL } from "../components/+add-cluster";
+import { clusterSettingsURL } from "../components/+cluster-settings";
+import { extensionsURL } from "../components/+extensions";
+import { landingURL } from "../components/+landing-page";
+import { preferencesURL } from "../components/+preferences";
+import { clusterViewURL } from "../components/cluster-manager/cluster-view.route";
+import { LensProtocolRouterRenderer } from "./router";
+import { navigate } from "../navigation/helpers";
+
+export function bindProtocolAddRouteHandlers() {
+  LensProtocolRouterRenderer
+    .getInstance<LensProtocolRouterRenderer>()
+    .addInternalHandler("/preferences", ({ search: { highlight }}) => {
+      navigate(preferencesURL({ fragment: highlight }));
+    })
+    .addInternalHandler("/landing", () => {
+      navigate(landingURL());
+    })
+    .addInternalHandler("/", () => {
+      navigate(landingURL());
+    })
+    .addInternalHandler("/cluster", () => {
+      navigate(addClusterURL());
+    })
+    .addInternalHandler("/cluster/:clusterId", ({ pathname: { clusterId } }) => {
+      navigate(clusterViewURL({ params: { clusterId } }));
```

Comment:
This should change the workspace, if necessary

Reply:
true

Reply:
I'll do these in a followup PR.